api/src/main/java/net/kyori/adventure/text/serializer/ComponentSerializer.java

```diff
@@ -74,6 +75,7 @@ public interface ComponentSerializer<I extends Component, O extends Component, R
    * @since 4.8.0
    */
   @Contract(value = "!null -> !null; null -> null", pure = true)
+  @Override
   default @Nullable O deserializeOrNull(final @Nullable R input) {
     return this.deserializeOr(input, null);
```

Comment:
```suggestion
    return ComponentDecoder.super.deserializeOr(input, null);
```

Reply:
I disagree with this suggestion, if an implementation overrides `deserializeOr` (and/or the other methods where `ComponentDecoder.super.blah` is called) the call will invoke the default method in the superinterface, not the overridden one as was the case before. IMO the default methods in ComponentSerializer inherited from ComponentDecoder can be removed entirely, it is not a breaking change, or alternatively left as-is

Reply:
Would personally prefer to remove them. Same should be done with the one from Encoder.
What do you think of that?

Reply:
what about:

```suggestion
    return ComponentDecoder.super.deserializeOrNull(input);
```

(having not looked closely at the context)

I'd like to keep the overrides, just so we can maintain the differing @since JD on them for people trying to work with older versions.

Reply:
If you all agreed to one solution I'm happy to change it that way.

Reply:
That should be fine, yeah, each one `super`ing to itself